frontend/src/test/java/starter/project/frontend/CarControllerTest.java

```diff
@@ -1,21 +1,56 @@
 package starter.project.frontend;
 
-import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.web.client.TestRestTemplate;
-import org.springframework.boot.web.server.LocalServerPort;
-import org.springframework.http.ResponseEntity;
+import starter.project.domain.Car;
+import starter.project.domain.Doors;
+import starter.project.domain.Engine;
+import starter.project.domain.Wheels;
+import java.util.ArrayList;
+import java.util.List;
+import static org.junit.Assert.assertEquals;
 
-import java.net.URL;
-import static org.junit.jupiter.api.Assertions.*;
-import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
 public class CarControllerTest {
-    @LocalServerPort
-    private int port;}
-//    @Autowired
-//    //private TestRestTemplate restTemplate
+
+    CarController controller = new CarController();
+
+    @Test
+    public void newCarsAAreAddedToMap() {
+
+        Car car = Car.builder().id(15).build();
+        controller.addCar(car);
+        assertEquals(1, controller.map.size());
+        Car secondCar = Car.builder().id(13).build();
+        controller.addCar(secondCar);
+        System.out.println(controller.map);
+        assertEquals(2, controller.map.size());
+    }
+    @Test
+    public void listOfCarsShouldbeAddedToMap(){
+        Car car = Car.builder().id(22).build();
+        Car secondCar = Car.builder().id(33).build();
+        Car thirdCar = Car.builder().id(19).build();
+        List<Car> list = new ArrayList<>();
+        list.add(car);
+        list.add(secondCar);
+        list.add(thirdCar);
+        List<Integer> idList = new ArrayList<>();
+        idList.add(22);
+        idList.add(33);
+        idList.add(19);
+        controller.createCarList(list);
+        assertEquals(list,controller.getBatchCars(idList));
+    }
+    @Test
+    public void verifyCarHasCorrectFields(){
+        Car car = Car.builder().id(22).doors(Doors.builder().build()).name("rover").engine(Engine.builder().build()).wheels(Wheels.builder().build()).build();
+        Car mockCar = Car.builder().id(22).doors(Doors.builder().build()).name("rover").engine(Engine.builder().build()).wheels(Wheels.builder().build()).build();
+        controller.addCar(car);
```

Comment:
A nice thing to have would be avoiding these `addCar` calls. Perhaps we can have a way to initialise the controller with some dummy data so that we are just testing what we say we are. I.e. if the test is `carInMapShouldReturn` then we don't want to rely on being able to add to the map, that's an assumption already made.

In the scope of this project however, it's probably ok. See what others say.